Observer/CheckoutObserver.php

```diff
@@ -92,6 +92,18 @@ public function afterSaveOrder(Varien_Event_Observer $event)
         }
     }
 
+    public function onShippingUpdate(Varien_Event_Observer $event)
+    {
+        /** @var Mage_Sales_Model_Quote $quote */
+        $quote = $event->getQuote();
+        try {
+            Bold_CheckoutPaymentBooster_Service_Order_Hydrate::hydrate($quote);
+        } catch (Mage_Core_Exception $e) {
+            Mage::log($e->getMessage(), Zend_Log::CRIT);
+            Mage::throwException(Mage::helper('core')->__('Bold update failed.'));
+        }
```

Comment:
A condition needs to be added to verify that the order was placed through Bold Booster, otherwise this could cause issues with normal orders. This may also adversely affect the Express Pay functionality, so we'll need to verify that it does not.

Reply:
I've added a check for the public order ID. I'm not sure if there's a better way to determine if this is a Bold Booster order.

With the Express Pay, I guess what could happen is that a the callback from the PayPal modal could trigger an update on the platform backend. Would that result in the `checkout_controller_onepage_save_shipping_method` event being fired?

Reply:
Yes, because we're calling the same AJAX endpoint for saving the shipping information as the one used in the checkout. We could possibly set a request parameter to denote it as coming from Express Pay and then do something like this in the Observer:

```php
$request = $observer->getEvent()->getRequest();

if ($request->getParam('source') === 'expresspay') {
    return;
}
```

Reply:
Yeah, that makes sense to me.